### PR TITLE
Add workarounds to IDE analyzers for missing IOperation/CFG support for using declarations

### PR DESCRIPTION
**Pull request by mavasani**
Workaround for #32100
1. https://github.com/dotnet/roslyn/commit/a08f86e3d5ec1701fc6ccd3c0e3538341832c2ad: Workaround in dispose analyzer
2. https://github.com/dotnet/roslyn/commit/b0a9514603afde201b10caf990e035fc5d92a2f1: Workaround in unused parameter/value analyzer

### CODE REVIEW

**Comment by mavasani** [comment]
@ivanbasov @tmat - seems like integration tests are consistently failing in master branch for an NRE in EnC integration tests:

https://dev.azure.com/dnceng/public/_build/results?buildId=237952&view=ms.vss-test-web.build-test-results-tab

```
Server stack trace: 
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.InvokeOnUIThread(Action`1 action)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.ExecuteOnActiveView(Action`1 action)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.ReplaceText(String oldText, String newText)
   at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
   at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.ReplaceText(String oldText, String newText)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.OutOfProcess.Editor_OutOfProc.ReplaceText(String oldText, String newText) in /_/src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs:line 141
   at Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicEditAndContinue.LocalsWindowUpdatesCorrectlyDuringEnC() in /_/src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicEditAndContinue.cs:line 299
----- Inner Stack Trace -----
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.InvokeOnUIThread(Action`1 action)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.ExecuteOnActiveView(Action`1 action)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.PlaceCaret(String marker, Int32 charsOffset, Int32 occurrence, Boolean extendSelection, Boolean selectBlock)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.SelectText(String text)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.Editor_InProc.<>c__DisplayClass19_0.<ReplaceText>b__0(IWpfTextView view)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c__DisplayClass13_0.<GetExecuteOnActionViewCallback>b__0(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.<>c__DisplayClass6_0.<<InvokeOnUIThread>b__0>d.MoveNext()
----- Inner Stack Trace -----
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c__DisplayClass8_0.<PlaceCaret>b__0(IWpfTextView view)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.TextViewWindow_InProc.<>c__DisplayClass13_0.<GetExecuteOnActionViewCallback>b__0(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess.InProcComponent.<>c__DisplayClass6_0.<<InvokeOnUIThread>b__0>d.MoveNext()
```

File: src/Features/Core/Portable/DisposeAnalysis/DisposeObjectsBeforeLosingScopeDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -103,6 +103,13 @@ protected override void InitializeWorker(AnalysisContext context)
                     ComputeDiagnostics(disposeDataAtExit, notDisposedDiagnostics, mayBeNotDisposedDiagnostics,
                         disposeAnalysisResult, pointsToAnalysisResult);
 
+                    if (disposeAnalysisResult.ControlFlowGraph.OriginalOperation.HasAnyOperationDescendant(o => o.Kind == OperationKind.None))
```
Comment (dibarbet) [inline src/Features/Core/Portable/DisposeAnalysis/DisposeObjectsBeforeLosingScopeDiagnosticAnalyzer.cs]: could this bail out earlier at the beginning of the try block or does this check depend on the result of ComputeDiagnostics?

Comment (mavasani) [reply src/Features/Core/Portable/DisposeAnalysis/DisposeObjectsBeforeLosingScopeDiagnosticAnalyzer.cs]: we can bail out earlier too. However, we will hopefully revert this soon in dev16.3 branch, which is the targeted release for #32100

**Comment by mavasani** [comment]
@jinujoseph @vatsalyaagrawal for approval

**Comment by jinujoseph** [comment]
@genlu could you review this as well pls

**Comment by mavasani** [comment]
Thanks @genlu @CyrusNajmabadi @dibarbet

**Pull request merged**